### PR TITLE
Small fixes from living unusually

### PR DESCRIPTION
**Pull request by nwf**
Riding around on a POWER Linux box brings out corner cases.

### CODE REVIEW

File: src/mem/chunkmap.h [hunk 1]
```diff
@@ -47,12 +59,14 @@ namespace snmalloc
 // Use flat map is under a single node.
 #  define SNMALLOC_MAX_FLATPAGEMAP_SIZE PAGEMAP_NODE_SIZE
```
Comment (mjp41) [inline src/mem/chunkmap.h:60]: Would it be better to change 
```
#  define SNMALLOC_MAX_FLATPAGEMAP_SIZE PAGEMAP_NODE_SIZE
```
to 
```
#  define SNMALLOC_MAX_FLATPAGEMAP_SIZE \
  pal_supports<LazyCommit> ? .... : PAGEMAP_NODE_SIZE
```
Where we choose `...` suitably.

Comment (nwf) [reply src/mem/chunkmap.h:60]: With default chunk size (i.e., neither `USE_SMALL_CHUNKS` nor `USE_LARGE_CHUNKS`), `SUPERSLAB_BITS` is 20.  With a 48-bit VA, that means a flat chunkmap pagemap is 256 MiB.  Is that a reasonable choice for `...`?

Comment (mjp41) [reply src/mem/chunkmap.h:60]: Probably, I would say going above 2^32 causes a lot of issues, so below that maybe 1GiB is the max?

Comment (nwf) [reply src/mem/chunkmap.h:60]: Permitting 512M will break `src/test/func/memory/memory.cc`; we could adjust the test, tho'.

Comment (mjp41) [reply src/mem/chunkmap.h]: Either is fine.  Just document the choice.

**Review by mjp41 — APPROVED** [review]
LGTM

**Pull request merged**